client/utils/account-fees.tsx

```diff
@@ -90,14 +90,19 @@ export const formatMethodFeesTooltip = (
 ): JSX.Element => {
 	if ( ! accountFees ) return <></>;
 	const currentBaseFee = getCurrentBaseFee( accountFees );
+	// If the current fee doesn't have a fixed or percentage rate, use the base fee's rate. Eg. when there is a promotional discount fee applied. Use this to calculate the total fee too.
+	const currentFeeWithBaseFallBack = currentBaseFee.percentage_rate
```

Comment:
Do we need to check if `fixed_rate` exists in the `currentBaseFee` as well? Or is it enough to check the `percentage_rate`?

Reply:
The scenarios I thought of seeing the existing fees were 

1. Custom fees - here we have a `percentage_rate` (for sure) and `fixed_rate` (optionally) within the discount fee object ( that gets selected as the currentBaseFee)  - example - 2ecc4-pb/raw/

2. Promotional fees - the ones i have seen, currently, only have a discount percentage mentioned, without base percentage_rate, nor fixed fees. - example - 2ecc5-pb/raw/

I was unable to find a case where we do not have percentage fee and have fixed fee. Hence thought the presence or absence of percentage_rate  would be a good benchmark to decide whether to use the base fee or the discount fee object data. 

If I were to account for the fixed fee, I am wondering what would be the logic for situations where there is only a `fixed_rate`, or only `percentage_rate` mentioned in the discount fee object. Should we be falling back on Base fee rates only if both `fixed_rate` and `percentage_rate` are absent, or either.. ( sorry if I am unable to explain this in a concise way)

Reply:
Thanks for the detailed explanation @nagpai.

> I was unable to find a case where we do not have percentage fee and have fixed fee. Hence thought the presence or absence of percentage_rate would be a good benchmark to decide whether to use the base fee or the discount fee object data.

This makes sense to me as well. Marking conversation as resolved